### PR TITLE
added new functions for getting cgmlst and miru cluster

### LINKED ISSUE #5: functions to extract miru and cgmlst cluster id
we need functions that return either a miru or a cgmlst cluster given a sample id.

### PR DESCRIPTION
**Pull request by sherrie9**

### CODE REVIEW

File: tb_db/crud.py [hunk 1]
```diff
@@ -341,3 +343,128 @@ def create_miru_profiles(db: Session, miru_profiles_by_sample_id: dict[str, obje
...
+        Sample.sample_id == sample_id
+    )
+
+    miru_ids = []
+    for row in query_result:
+        miru_ids.append(utils.row2dict(row))
+    
+    for item in miru_ids:
+        if item['valid_until'] is None:
+            miru_cluster_id = item['miru_cluster_id']
```
Comment (dfornika) [inline tb_db/crud.py]: Could we clarify the variable names in this section? The items in the `query_result` are `Samples`. So when we iterate over them, convert them to dictionaries and store them in a new list, that list should probably be called `samples` or `sample_dicts`?

File: tb_db/crud.py [hunk 2]
```diff
@@ -341,3 +343,128 @@ def create_miru_profiles(db: Session, miru_profiles_by_sample_id: dict[str, obje
...
+    if sample_id not in existing_sample_ids:
+        logging.warning('cannot add cgmlst cluster for a sample that does not exist...')
+        return None
+
+    else:
+        select_sample_stmt = select(Sample).where(and_(Sample.sample_id == sample_id, Sample.valid_until == None))
+        sample = db.scalars(select_sample_stmt).one()
+        sample.cgmlst_cluster_id = db_cgmlst_cluster.id
+        db.commit()
+        return sample
```
Comment (dfornika) [inline tb_db/crud.py:448]: We may need to call db.refresh() to update the `sample.id`. See for example:

https://github.com/BCCDC-PHL/tb-db/blob/e0e8beb99417982503741ad9a25c10d2987991c9/tb_db/crud.py#L245-L250

Could you add a unit test to check how the `sample.id` behaves when a sample is added to a cgMLST cluster? I'm not sure if the existing record will be invalidated, so the valid `sample.id` may be changed when it's added to the cgMLST cluster. Does the returned value have the latest valid `sample.id`?

**Review by dfornika — CHANGES_REQUESTED** [review]
Looking good. Just a couple changes needed.

File: tb_db/crud.py (hunk 2 above)
Comment (sherrie9) [reply tb_db/crud.py:448]: I added two lines in test_crud.py line #82 --> self.assertEqual(created_sample.id,1) and line #90-->  self.assertEqual(result.id, 2), which passed the test. it showed that when we first create the sample, it is assigned an id 1, after we add this sample to a cgmlst cluster, the record with id 1 got invalidated and now it is given a new id, 2.

**Review by dfornika — APPROVED** [review]
Great!

**Pull request merged**